src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+
+Replicate the shell commands below on each node using your preferred deployment model. Be sure to edit the values of the flags passed to [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}) as needed for your environment.
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ export COCKROACH_LOG_MAX_SYNC_DURATION=40s
+$ export COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT=40s
+$ cockroach start --certs-dir certs --listen-addr=:26257 --http-addr=:26258 --advertise-addr=10.150.0.54:26257 --join=34.48.145.131:26257 --store path=/mnt/data1/cockroach,attrs=store1:node1:node1store1 --cache=25% --locality=cloud=gce,region=us-east4,zone=us-east4-a,rack=0 --log-config-file=logs.yaml --wal-failover=path=/mnt/data2/cockroach
+~~~
+
+Notice the flags passed to `cockroach start`:
+
+- `--wal-failover`'s value is `path=/mnt/data2/cockroach` since that is the single failover disk
+- `--log-config-file` is `logs.yaml`
+- and the data store (store 1) path in `--store` is `/mnt/data1/cockroach`
+
+<a name="multi-store-config"></a>
+
+### Provisioning a multi-store cluster for WAL failover
+
+In this section, we will explain how to provision a multi-store cluster and configure it for WAL failover.
+
+#### 1. Create the cluster
+
+Provision a 3 node cluster with 4 SSDs for each node. Deploy each node to a different region (e.g. [in GCE](https://cloud.google.com/compute/docs/regions-zones), `us-east4-a`, `us-west2-b`, `us-central1-c`). Be sure to [create a separate volume for each SSD](https://cloud.google.com/compute/docs/disks).
+
+#### 2. Stage the cluster
+
+[Install CockroachDB]({% link {{ page.version.version }}/install-cockroachdb-linux.md %}) on each node.
+
+#### 3. Log Configuration for WAL Failover
+
+If you are logging to a file-based sink, create a `logs.yaml` file. Later on, you will pass this file to `cockroach start` when starting each node in the cluster. Not doing so will negate the positive impact of enabling WAL failover, because writes to the diagnostic logs may block indefinitely during the disk stall, effectively stalling the node.
+
+{% include {{ page.version.version }}/wal-failover-log-config.md %}
+
+Next transfer this file to the nodes on the cluster.
+
+For more information about how to configure CockroachDB logs, see [Configure logs]({% link {{ page.version.version }}/configure-logs.md %}).
+
+#### 4. Starting the multi-store cluster with WAL failover enabled
+
+To enable WAL failover when we start the cluster,  we can either pass the [`--wal-failover=path={ path-to-my-side-disk-for-wal-failover }` to `cockroach start`]({% link {{ page.version.version }}/cockroach-start.md%}#enable-wal-failover)
+
+ *OR*
+
+set [the environment variable `COCKROACH_WAL_FAILOVER=path={ path-to-my-side-disk-for-wal-failover }`]({% link {{ page.version.version }}/cockroach-start.md%}#enable-wal-failover) before starting the cluster.
```

Comment:
and `COCKROACH_WAL_FAILOVER=among-stores`

Reply:
fixed!